composeApp/src/commonMain/kotlin/com/overdrive/cruiser/views/SpotDetailView.kt

```diff
@@ -38,158 +48,226 @@ import com.overdrive.cruiser.models.BookingStatus
-                                bookingStatus = BookingStatus.PENDING,
-                                vehicleLicensePlate = "1234"
+                        Row {
+                            Icon(
+                                modifier = Modifier.size(24.dp),
+                                imageVector = Icons.Filled.Share,
+                                contentDescription = "Share",
+                                tint = Color.Black
                             )
-                            scope.launch {
-                                bookingEndpoint.create(booking)
+                            Spacer(modifier = Modifier.size(16.dp))
+                            Icon(
+                                modifier = Modifier.size(24.dp),
+                                imageVector = Icons.Filled.FavoriteBorder,
+                                contentDescription = "Bookmark",
+                                tint = Color.Black // Optional: set the color of the icon
+                            )
+                        }
+                    }
+
+                    Column(modifier = Modifier.padding(16.dp)) {
+                        SpotOnField(modifier = Modifier.fillMaxWidth()) {
+                            Row(
+                                modifier = Modifier.padding(16.dp),
+                                verticalAlignment = Alignment.CenterVertically
+                            ) {
+                                Text(spot.address)
+                            }
+                        }
+                        SpotOnField(modifier = Modifier.fillMaxWidth()) {
+                            Row(
+                                modifier = Modifier.padding(16.dp),
+                                verticalAlignment = Alignment.CenterVertically
+                            ) {
+                                Text("$5/hour")
+                                Spacer(modifier = Modifier.weight(1f))
+                                Text("$40/day")
+                            }
+                        }
+                        SpotOnField(modifier = Modifier.fillMaxWidth()) {
+                            Row(
+                                modifier = Modifier.padding(16.dp),
+                                verticalAlignment = Alignment.CenterVertically
+                            ) {
+                                Text("226 total bookings")
+                            }
+                        }
+
+                        Spacer(modifier = Modifier.weight(1f))
+
+                        Text("Book", fontSize = 24.sp, fontWeight = FontWeight.Bold)
+
+                        val tabs = listOf("Hourly", "Daily", "Schedule")
+                        var hoursToBook by remember { mutableStateOf("1") }
+                        SpotOnTabRow(
+                            tabs = tabs,
+                            selectedTabIndex = selectedTab,
+                            { selectedTab = it }
```

Comment:
This looks a lil sus do we actually need all these variables? I designed it just to take a when() with 0..len

Reply:
This was just so it maintains the tab when i'm switching to the date picker view, there might be a better way to do this but it was just defaulting back to the first tab every time

Reply:
I think this makes more sense to encapsulate in the TabView if possible, rather than spill back into the parents view.

Reply:
Might address this in a later pr, can't seem to get it to work with only having the state in the tab view.